src/Controller/Api/JoinProgrammeController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Api;
+
+use App\Repository\ProgrammeRepository;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+class JoinProgrammeController extends AbstractController
+{
+    private ProgrammeRepository $programmeRepository;
+
+    private UserRepository $userRepository;
+
+    public function __construct(
+        ProgrammeRepository $programmeRepository,
+        UserRepository $userRepository
+    ) {
+        $this->programmeRepository = $programmeRepository;
+        $this->userRepository = $userRepository;
+    }
+
+    /**
+     * @Route (path="/api/programmes/join/{programmeId}", methods={"POST"})
+     */
+    public function index(Request $request, int $programmeId): Response
+    {
+        $loggedInUserId = $this->getUser()->getId();
+        $userToBeJoinedId = json_decode($request->getContent())->id ?? null;
+        $programmeToBeJoined = $this->programmeRepository->findOneBy(['id' => (int)$programmeId]);
+
+        if (null !== $userToBeJoinedId && $loggedInUserId !== $userToBeJoinedId) {
+            $this->denyAccessUnlessGranted('ROLE_ADMIN');
+
+            try {
+                $this->userRepository->joinAProgramme($userToBeJoinedId, $programmeToBeJoined);
+            } catch (NotFoundHttpException $e) {
```

Comment:
your repository throws a not found HTTP exception?! this sound bad, repository should not throw any exception that has HTTP in its name